dojo_plugin/api/v1/docker.py

```diff
@@ -327,16 +326,16 @@ def start_challenge(user, dojo_challenge, practice, *, as_user=None):
     as_user = as_user or user
     docker_client = docker.from_env()
     remove_container(docker_client, user)
-
     umount_existing_overlay(user)
-    setup_home(user)
-    mounts = [("/home/hacker", HOST_HOMES / "nosuid" / str(user.id))]
+
+    setup_home(as_user)
+    mounts = [("/home/hacker", HOST_HOMES_NOSUID / str(as_user.id))]
     if as_user != user:
-        setup_home(as_user)
+        setup_home(user)
```

Comment:
Just curious, why did you end up swapping user and as_user here? If I'm understanding correctly either order works so it shouldn't matter.

Reply:
It doesn't technically make a difference as you point out. But in the same way that `insert_challenge` takes an `as_user`, the "`as_user`"  has the "priority", by default when we're setting up a home, it's the `as_user`'s home, and then if there's also a separate `user`, we need to set up for them also.